src/curses/ui_call_flow.c

```diff
@@ -641,37 +641,19 @@ call_flow_draw_rtp_stream(ui_t *ui, call_flow_arrow_t *arrow, int cline)
     callid = stream->media->msg->call->callid;
     msg_src = stream->media->msg->packet->src;
     msg_dst = stream->media->msg->packet->dst;
-    stream_src = stream->src;
-    stream_src.port = 0;
-    stream_dst = stream->dst;
-    stream_dst.port = 0;
-
-    // Get origin column for this stream.
-    // If we share the same Address from its setup SIP packet, use that column instead.
-    if (!strcmp(stream->src.ip, msg_src.ip)) {
-        column1 = call_flow_column_get(ui, callid, msg_src);
-    } else if (!strcmp(stream->src.ip, msg_dst.ip)) {
-        column1 = call_flow_column_get(ui, callid, msg_dst);
-    } else {
-        column1 = call_flow_column_get(ui, 0, stream_src);
-    }
 
-    // Get destination column for this stream.
-    // If we share the same Address from its setup SIP packet, use that column instead.
-    if (!strcmp(stream->dst.ip, msg_dst.ip)) {
-        column2 = call_flow_column_get(ui, callid, msg_dst);
-    } else if (!strcmp(stream->dst.ip, msg_src.ip)) {
-        column2 = call_flow_column_get(ui, callid, msg_src);
-    } else {
-        column2 = call_flow_column_get(ui, 0, stream_dst);
-    }
+    // Get call inititator source and destination address
+    src = stream->media->msg->call->cstart_msg->packet->src;
+    dst = stream->media->msg->call->cstart_msg->packet->dst;
```

Comment:
With this change, starting and ending columns for all rtp streams of the dialog are the same because their position is based in "the first message" of the call. When using proxies that act as media relay, this may not be true as shown in the screens below.